src/Server.cpp

```diff
@@ -148,6 +148,16 @@ static void applyDefaults(UA_ServerConfig* config) {
 
 Server::Server(uint16_t port, ByteString certificate)
     : connection_(std::make_shared<Connection>(*this)) {
+    init(port, std::move(certificate));
+}
+
+Server::Server(Logger logger, uint16_t port, ByteString certificate)
+    : connection_(std::make_shared<Connection>(*this)) {
+    setLogger(std::move(logger));
+    init(port, std::move(certificate));
+}
+
+void Server::init(uint16_t port, ByteString certificate) {
```

Comment:
This only state the member function accesses in the `ServerConfig*`. Lets make it a static free function and call it in both constructors. E.g.:

```cpp
inline static setMinimal(ServerConfig* config, uint16_t port, const ByteString& certificate) {
   // ...
}
```

And just call it with `setMinimal(getConfig(this), ...)`.

Reply:
makes perfect sense  👍

However, should the function really be named `setMinimal` ?

Besides the `UA_ServerConfig_setMinimal()`, both constructor currently also set some default subscription settings (`applyDefaults()`) and configure default access control (`setAccessControl()`).

I would either only put the `UA_ServerConfig_setMinimal` in the static function and leave the other two calls in both constructors, or maybe name the function that does all of that e.g. `initializeServer()`. 
What do you think?

Reply:
Naming is hard 🤔 But you are right, `setMinimal` is a weak choice.
Maybe we go with `initializeServer` or `initServerConfig`. If necessary, we later overload this e.g. for other constructors (`UA_ServerConfig_setDefaultWithSecurityPolicies`).

Reply:
done 👍

Reply:
Looks good! 👍